### PR TITLE
ML Vision iOS

### PR DESCRIPTION
**Pull request by Ehesp**

### CODE REVIEW

File: packages/ml-vision/ios/RNFBMLVision/RNFBMLVisionModule.m [hunk 1]
```diff
@@ -15,24 +15,21 @@
...
-  - (dispatch_queue_t)methodQueue {
-    return dispatch_get_main_queue();
-  }
+RCT_EXPORT_MODULE();
 
 #pragma mark -
-#pragma mark Firebase Mlkit Methods
+#pragma mark Firebase ML Kit Vision Methods
+
+// TODO react methods here
```
Comment (Ehesp) [inline packages/ml-vision/ios/RNFBMLVision/RNFBMLVisionModule.m]: Delete this file?

File: packages/ml-vision/ios/RNFBMLVision/RNFBMLVisionTextRecognizerModule.m [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+- (NSArray *)getVisionTextBlocksList:(NSArray <FIRVisionTextBlock *> *)blocks {
+  NSMutableArray *blockListFormatted = [[NSMutableArray alloc] init];
+
+  for (FIRVisionTextBlock *block in blocks) {
+    NSMutableDictionary *textBlockFormatted = [[NSMutableDictionary alloc] init];
+
+    textBlockFormatted[@"text"] = block.text;
+    textBlockFormatted[@"confidence"] = block.confidence;
+    textBlockFormatted[@"recognizedLanguages"] = [self getLanguageCodesList:block.recognizedLanguages];
+//    textBlockFormatted[@"cornerPoints"] = [RNFBMLVisionCommon visionPointsToArray:block.cornerPoints];
```
Comment (Ehesp) [inline packages/ml-vision/ios/RNFBMLVision/RNFBMLVisionTextRecognizerModule.m]: Corner points to point array issue

File: packages/ml-vision/ios/RNFBMLVision/RNFBMLVisionTextRecognizerModule.m [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+  NSMutableArray *lineListFormatted = [[NSMutableArray alloc] init];
+
+  for (FIRVisionTextLine *line in lines) {
+    NSMutableDictionary *lineFormatted = [[NSMutableDictionary alloc] init];
+
+    lineFormatted[@"boundingBox"] = [RNFBMLVisionCommon rectToIntArray:line.frame];
+    lineFormatted[@"text"] = line.text;
+    lineFormatted[@"confidence"] = line.confidence;
+    lineFormatted[@"recognizedLanguages"] = [self getLanguageCodesList:line.recognizedLanguages];
+//    lineFormatted[@"cornerPoints"] = [RNFBMLVisionCommon visionPointsToArray:line.cornerPoints];
```
Comment (Ehesp) [inline packages/ml-vision/ios/RNFBMLVision/RNFBMLVisionTextRecognizerModule.m]: Corner points to point array issue

File: packages/ml-vision/ios/RNFBMLVision/RNFBMLVisionTextRecognizerModule.m [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+  NSMutableArray *elementsListFormatted = [[NSMutableArray alloc] init];
+
+  for (FIRVisionTextElement *element in elements) {
+    NSMutableDictionary *elementFormatted = [[NSMutableDictionary alloc] init];
+
+    elementFormatted[@"boundingBox"] = [RNFBMLVisionCommon rectToIntArray:element.frame];
+    elementFormatted[@"text"] = element.text;
+    elementFormatted[@"confidence"] = element.confidence;
+    elementFormatted[@"recognizedLanguages"] = [self getLanguageCodesList:element.recognizedLanguages];
+//    elementFormatted[@"cornerPoints"] = [RNFBMLVisionCommon visionPointsToArray:element.cornerPoints];
```
Comment (Ehesp) [inline packages/ml-vision/ios/RNFBMLVision/RNFBMLVisionTextRecognizerModule.m]: Corner points to point array issue

File: packages/ml-vision/ios/RNFBMLVision/RNFBMLVisionBarcodeDetectorModule.m [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+          @"code": errorCodeMessageArray[0],
+          @"message": errorCodeMessageArray[1],
+      }];
+      return;
+    }
+
+    FIRVisionImage *visionImage = [[FIRVisionImage alloc] initWithImage:image];
+    FIRVision *vision = [FIRVision vision];
+
+    FIRVisionBarcodeDetectorOptions *options = [[FIRVisionBarcodeDetectorOptions alloc] initWithFormats: FIRVisionBarcodeFormatQRCode | FIRVisionBarcodeFormatAztec];
```
Comment (Ehesp) [inline packages/ml-vision/ios/RNFBMLVision/RNFBMLVisionBarcodeDetectorModule.m]: How do we convert a list of ints (from JS) into this type? Left as per the example for now.

File: packages/ml-vision/ios/RNFBMLVision/RNFBMLVisionBarcodeDetectorModule.m [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+- (NSArray *)getBarcodesList:(NSArray<FIRVisionBarcode *> *)barcodes {
+  NSMutableArray *barcodeListFormatted = [[NSMutableArray alloc] init];
+
+  for (FIRVisionBarcode *barcode in barcodes) {
+    NSMutableDictionary *formattedBarcode = [[NSMutableDictionary alloc] init];
+
+    formattedBarcode[@"boundingBox"] = [RNFBMLVisionCommon rectToIntArray:barcode.frame];
+    formattedBarcode[@"cornerPoints"] = @[]; // TODO NSValue to Point
+    formattedBarcode[@"format"] = @(barcode.format);
+    formattedBarcode[@"valueType"] = @(barcode.valueType); // TODO correct?
```
[the remark this answers is not in the document]
Comment (Salakar) [reply packages/ml-vision/ios/RNFBMLVision/RNFBMLVisionBarcodeDetectorModule.m]: Fixed in commit I just pushed from your Mac

File: packages/ml-vision/ios/RNFBMLVision/RNFBMLVisionBarcodeDetectorModule.m [hunk 7]
```diff
@@ -0,0 +1,239 @@
...
+- (NSDictionary *)getURLMap:(FIRVisionBarcodeURLBookmark *)url {
+  return @{
+    @"title": url.title,
+    @"url": url.url,
+  };
+}
+
+- (NSDictionary *)getWiFiMap:(FIRVisionBarcodeWiFi *)wifi {
+  return @{
+    @"encryptionType": @(wifi.type), // TODO working?
```
[the remark this answers is not in the document]
Comment (Salakar) [reply packages/ml-vision/ios/RNFBMLVision/RNFBMLVisionBarcodeDetectorModule.m]: Fixed in commit I just pushed from your Mac

File: packages/ml-vision/ios/RNFBMLVision/RNFBMLVisionCommon.m [hunk 8]
```diff
@@ -0,0 +1,216 @@
...
+  if ([@"CalendarEvent" isEqualToString:valueType]) {
+    return @11;
+  }
+  if ([@"DriversLicense" isEqualToString:valueType]) {
+    return @12;
+  }
+  return @-1;
+}
+
++ (NSArray *)visionPointsToArray:(NSArray <NSValue *> *_Nullable)points {
```
[the remark this answers is not in the document]
Comment (Salakar) [reply packages/ml-vision/ios/RNFBMLVision/RNFBMLVisionCommon.m]: Fixed in commit I just pushed from your Mac

File: packages/ml-vision/ios/RNFBMLVision/RNFBMLVisionModule.m (hunk 1 above)
Comment (Salakar) [reply packages/ml-vision/ios/RNFBMLVision/RNFBMLVisionModule.m]: Fixed in commit I just pushed from your Mac

File: packages/ml-vision/ios/RNFBMLVision/RNFBMLVisionTextRecognizerModule.m (hunk 2 above)
Comment (Salakar) [reply packages/ml-vision/ios/RNFBMLVision/RNFBMLVisionTextRecognizerModule.m]: Fixed in commit I just pushed from your Mac

File: packages/ml-vision/ios/RNFBMLVision/RNFBMLVisionTextRecognizerModule.m (hunk 3 above)
Comment (Salakar) [reply packages/ml-vision/ios/RNFBMLVision/RNFBMLVisionTextRecognizerModule.m]: Fixed in commit I just pushed from your Mac

File: packages/ml-vision/ios/RNFBMLVision/RNFBMLVisionTextRecognizerModule.m (hunk 4 above)
Comment (Salakar) [reply packages/ml-vision/ios/RNFBMLVision/RNFBMLVisionTextRecognizerModule.m]: Fixed in commit I just pushed from your Mac

File: packages/ml-vision/ios/RNFBMLVision/RNFBMLVisionBarcodeDetectorModule.m [hunk 9]
```diff
@@ -0,0 +1,239 @@
...
+    @"address": email.address,
+    @"body": email.body,
+    @"subject": email.subject,
+  };
+}
+
+- (NSDictionary *)getPhoneMap:(FIRVisionBarcodePhone *)phone {
+  return @{
+    @"number": phone.number,
+    @"type": @(phone.type), // TODO working?
```
[the remark this answers is not in the document]
Comment (Salakar) [reply packages/ml-vision/ios/RNFBMLVision/RNFBMLVisionBarcodeDetectorModule.m]: Fixed in commit I just pushed from your Mac

File: packages/ml-vision/ios/RNFBMLVision/RNFBMLVisionBarcodeDetectorModule.m [hunk 10]
```diff
@@ -0,0 +1,239 @@
...
+}
+
+- (NSArray *)getBarcodesList:(NSArray<FIRVisionBarcode *> *)barcodes {
+  NSMutableArray *barcodeListFormatted = [[NSMutableArray alloc] init];
+
+  for (FIRVisionBarcode *barcode in barcodes) {
+    NSMutableDictionary *formattedBarcode = [[NSMutableDictionary alloc] init];
+
+    formattedBarcode[@"boundingBox"] = [RNFBMLVisionCommon rectToIntArray:barcode.frame];
+    formattedBarcode[@"cornerPoints"] = @[]; // TODO NSValue to Point
```
[the remark this answers is not in the document]
Comment (Salakar) [reply packages/ml-vision/ios/RNFBMLVision/RNFBMLVisionBarcodeDetectorModule.m]: Fixed in commit I just pushed from your Mac

File: packages/ml-vision/ios/RNFBMLVision/RNFBMLVisionBarcodeDetectorModule.m [hunk 11]
```diff
@@ -0,0 +1,239 @@
...
+  }
+  contactInfoFormatted[@"addresses"] = addresses;
+
+  return contactInfoFormatted;
+}
+
+- (NSDictionary *)getCalendarEventMap:(FIRVisionBarcodeCalendarEvent *)event {
+  return @{
+    @"description": event.description,
+    @"end": event.end, // TODO convert to string?
```
[the remark this answers is not in the document]
Comment (Salakar) [reply packages/ml-vision/ios/RNFBMLVision/RNFBMLVisionBarcodeDetectorModule.m]: Fixed in commit I just pushed from your Mac

File: packages/ml-vision/ios/RNFBMLVision/RNFBMLVisionBarcodeDetectorModule.m [hunk 12]
```diff
@@ -0,0 +1,239 @@
...
+    @"location": event.location,
+    @"organizer": event.organizer,
+    @"start": event.start, // TODO convert to string?
```
[the remark this answers is not in the document]
Comment (Salakar) [reply packages/ml-vision/ios/RNFBMLVision/RNFBMLVisionBarcodeDetectorModule.m]: Fixed in commit I just pushed from your Mac

**Pull request merged**